File: test-api-pkg/py/test_docs_troubleshooting.py
"""Mirrors website/public/guides/troubleshooting.md against `imbrace==1.0.4`
(PyPI) — API-key auth.

The doc shows error scenarios — we trigger the documented bug to verify
the SDK behaves as the doc claims:
  - `channel.list()` without `type` raises 400 (per doc)
  - `channel.list(type="web")` works
"""
from __future__ import annotations
import os, sys, json
from dotenv import load_dotenv
load_dotenv()
from imbrace import ImbraceClient

API_KEY = os.environ.get("IMBRACE_API_KEY"); ORG_ID = os.environ.get("IMBRACE_ORGANIZATION_ID")
GATEWAY = os.environ.get("IMBRACE_GATEWAY_URL", "https://app-gatewayv2.imbrace.co")
if not API_KEY or not ORG_ID:
    print("Missing creds"); sys.exit(1)

client = ImbraceClient(api_key=API_KEY, organization_id=ORG_ID, gateway=GATEWAY, timeout=8)
passed = failed = 0; failures: list[str] = []


def step(label, fn, expect_fail=False):
    global passed, failed
    sys.stdout.write(f"  - {label} ... "); sys.stdout.flush()
    try:
        r = fn(); summary = json.dumps(r, default=str)[:90] if r is not None else ""
        if expect_fail:
            print(f"unexpected pass {summary}"); failed += 1; failures.append(f"{label} -> unexpected pass")
        else:
            print(f"OK {summary}"); passed += 1
    except Exception as e:
        code = str(e)[:80]
        if expect_fail:
            print(f"OK (expected fail [{code}])"); passed += 1
        else:
            print(f"FAIL [{code}]"); failed += 1; failures.append(f"{label} -> {code}")


print(f"\n=== DOCS: guides/troubleshooting.md - auth: API KEY (PyPI imbrace==1.0.4) ===\n")

print("== §channel.list — doc claims missing `type` triggers 400 ==")
step("channel.list() without `type` — expected 400 per doc",
     lambda: client.channel.list(), expect_fail=True)
step("channel.list(type='web') — should work",
     lambda: client.channel.list(type="web"))

print(f"\n=== Summary (troubleshooting / API key) ===")
print(f"pass={passed}  fail={failed}  skip=0")
if failures:
    print("Failures:")
    for f in failures: print(f"  - {f}")
sys.exit(1 if failed > 0 else 0)

File: test-api-pkg/ts/test-docs-troubleshooting.ts
/**
 * Mirrors website/public/guides/troubleshooting.md against @imbrace/sdk@1.0.4
 * (npm) — API-key auth.
 *
 * troubleshooting.md is mostly Py-flavoured and covers .env / module-not-
 * found scenarios. The one cross-language SDK behaviour to verify is:
 *
 *   `channel.list()` without `type` → 400 per doc
 *   `channel.list({ type: "web" })` works
 */
import "dotenv/config"
import { ImbraceClient } from "@imbrace/sdk"

const apiKey         = process.env.IMBRACE_API_KEY
const organizationId = process.env.IMBRACE_ORGANIZATION_ID
const baseUrl        = process.env.IMBRACE_GATEWAY_URL ?? "https://app-gatewayv2.imbrace.co"

if (!apiKey || !organizationId) { console.error("Missing creds"); process.exit(1) }

const client = new ImbraceClient({ apiKey, organizationId, baseUrl, timeout: 8_000 })

let pass = 0, fail = 0
const fails: string[] = []

async function step(label: string, fn: () => Promise<unknown>, expectFail = false) {
  process.stdout.write(`  • ${label} ... `)
  try {
    const r = await fn()
    if (expectFail) { console.log(`unexpected pass: ${JSON.stringify(r ?? {}).slice(0, 60)}`); fail++; fails.push(`${label} → unexpected pass`) }
    else { console.log(`✓ ${JSON.stringify(r ?? {}).slice(0, 60)}`); pass++ }
  } catch (err: any) {
    const code = err?.statusCode ?? err?.message ?? "ERR"
    if (expectFail) { console.log(`✓ (expected fail [${code}])`); pass++ }
    else { console.log(`✗ [${code}]`); fail++; fails.push(`${label} → ${code}`) }
  }
}

console.log(`\n━━━ DOCS: guides/troubleshooting.md — auth: API KEY (npm @imbrace/sdk@1.0.4) ━━━\n`)
console.log("══ §channel.list — doc claims missing `type` triggers 400 ══")
await step("channel.list() — expected 400 per doc",
  () => client.channel.list({} as any), /* expectFail */ true)
await step("channel.list({ type: 'web' }) — should work",
  () => client.channel.list({ type: "web" } as any))

console.log(`\n━━━ Summary (troubleshooting / API key) ━━━`)
console.log(`pass=${pass}  fail=${fail}  skip=0`)
if (fails.length) { console.log("Failures:"); fails.forEach(f => console.log(`  - ${f}`)) }
process.exit(fail > 0 ? 1 : 0)
